### PR TITLE
Added balancer device

### PR DESCRIPTION
**Pull request by Nikita-V**
Balancing log messages between several outputs

### CODE REVIEW

File: lib/logstash-logger/device/balancer.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      def self.delegate_to_one(*methods)
+        methods.each do |m|
+          define_method(m) do |*args|
+            select_device(*args).send(m, *args)
+          end
+        end
+      end
+
+      def select_device(args)
+        devices[Zlib::crc32(args) % devices.count]
```
Comment (dwbutler) [inline lib/logstash-logger/device/balancer.rb]: What's the advantage of this over `devices.sample`? It seems to me like this way could potentially get unbalanced, sending more writes to some devices than others.

**Comment by dwbutler** [comment]
@Nikita-V Thanks for the contribution! I'd like to merge this. I just had a question about selecting a device via checksum vs just selecting one randomly.

**Comment by Nikita-V** [comment]
@dwbutler you're right. Random selection is better.

**Comment by dwbutler** [comment]
:+1:

**Comment by dwbutler** [comment]
Released in 0.11.0.

**Pull request merged**